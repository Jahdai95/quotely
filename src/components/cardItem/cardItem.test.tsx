import { render } from "@testing-library/react";
import { describe, test, vi, it, expect } from "vitest";
import CardItem from "./cardItem";
import { QuotesProvider } from "../../providers/quotes.provider";

describe("Should render CardItem component", () => {
  const getRender = () => {
    return render(
      <QuotesProvider>
        <CardItem
          quote="Solo viven aquellos que luchan"
          handleDelete={() => vi.fn()}
          id="01"
        />
      </QuotesProvider>
    );
  };

  it("Should macth snapshot", () => {
    const { container } = getRender();
    expect(container).toMatchSnapshot();
  });
});
