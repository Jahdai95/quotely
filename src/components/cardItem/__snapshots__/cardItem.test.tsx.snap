// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render CardItem component > Should macth snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-196bwzp-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-1bzm7am-MuiCardContent-root"
    >
      <p
        class="quicksand-200"
      >
        Solo viven aquellos que luchan
      </p>
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1svfwy9-MuiButtonBase-root-MuiIconButton-root"
        data-testid="button-delete-quote"
        id=":r0:"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Should render CardItem component > debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-196bwzp-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-1bzm7am-MuiCardContent-root"
    >
      <p
        class="quicksand-200"
      >
        Solo viven aquellos que luchan
      </p>
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1svfwy9-MuiButtonBase-root-MuiIconButton-root"
        data-testid="button-delete-quote"
        id=":r0:"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
